ambry-server/src/main/java/com.github.ambry.server/AmbryServerRequests.java

```diff
@@ -614,6 +615,10 @@ protected ServerErrorCode validateRequest(PartitionId partition, RequestOrRespon
       metrics.partitionReadOnlyError.inc();
       return ServerErrorCode.Partition_ReadOnly;
     }
+    if (requestType.equals(RequestOrResponseType.UndeleteRequest) && !serverConfig.serverHandleUndeleteRequestEnabled) {
+      metrics.temporarilyDisabledError.inc();
+      return ServerErrorCode.Temporarily_Disabled;
+    }
```

Comment:
Can we move this piece of code into `isRequestEnabled()` ?

Reply:
sure